grease-cli/tests/sanity/pass/dwarf-shape-recursive/test.c

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) Galois, Inc. 2025 */
+
+// Tests that heuristics are unneeded when using dwarf populated shapes to 
+// create a memory precondition that allows sum_list to execute.
+// Tests that the recursive type instantiation is bounded 
+
+
+struct llnode {
+  int x;
+  struct llnode *next;
+};
+
+int sum_list(struct llnode *in) {
+  struct llnode *curr = in;
+  int sum = 0;
+  while (curr) {
+    sum += curr->x;
+    curr = curr->next;
+  }
+  return sum;
+}
+
+// all: flags {"--symbol", "sum_list", "--enable-dwarf-preconditions", "--no-heuristics"}
+// x64: go(prog) 
+// x64: ok()
+
+// all: flags {"--symbol", "sum_list", "--enable-dwarf-preconditions", "--no-heuristics", "--type-unrolling-bound", "12"}
+// x64: go(prog) 
+// x64: ok()
+
+// all: flags {"--symbol", "sum_list", "--enable-dwarf-preconditions", "--no-heuristics", "--type-unrolling-bound", "0"}
+// x64: go(prog) 
+// x64: ok()
```

Comment:
The assertions here don't seem to meaningfully differ from the test for the default value, so I think the test could still pass even if this flag didn't do anything. It would be nice to test the generated shapes, as is done here:

https://github.com/GaloisInc/grease/pull/254/files#diff-f49829d9043ff039dfb0971ac8c722f5efb92871f4a6ab08e91082b9dbaf2e90R64